modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/di/RepositoryProviderModule.kt

```diff
@@ -28,12 +29,13 @@ class RepositoryProviderModule {
     @Provides
     @Singleton
     @DownloadOkHttpClient
-    fun downloadOkHttpClient(): OkHttpClient {
+    fun downloadOkHttpClient(networkInterceptors: Set<@JvmSuppressWildcards Interceptor>): OkHttpClient {
         val dispatcher = Dispatcher().apply {
             maxRequestsPerHost = 5
         }
         return OkHttpClient.Builder()
             .dispatcher(dispatcher)
+            .apply { networkInterceptors().addAll(networkInterceptors) }
```

Comment:
Sorry, I know this is a draft PR but I wonder if we shouldn't bother with monitoring the network performance of episode downloads as these come from podcast producer servers and not our own. iOS does this with Firebase Performance and it adds a lot of noise to the metrics.

<img width="1293" alt="Screenshot 2024-05-07 at 10 24 18 PM" src="https://github.com/Automattic/pocket-casts-android/assets/308331/fe25e473-494f-4125-b2b5-d1624a007a1a">

Reply:
That's a good point! WooCommerce Android has a similar concern when store owners connect to the stores hosted on their own hosting providers.

I agree, that it's not useful data in isolation. It is useful though in context of performance transactions. I mean instead of answering questions "which network queries are the slowest" we'll be able to answer questions "why loading podcasts is slow" - is it because of e.g. performance bottleneck of our app, or because slow http request to a 3rd party podcast server? If we don't introduce the http interceptor, we'll have "Missing instrumentation" in places when http happens and we won't be able to identify what was happening during this time.

Reply:
I'm happy to take your advice on including them if you recommend it.

Reply:
Gotcha, thanks. To visualize my point above, having [a sample transaction](https://a8c.sentry.io/performance/trace/5b82af1a0b2547a79026fbae92ae8a2c/?node=trace-root&statsPeriod=1h&timestamp=1715158865):
<img width="1490" alt="Screenshot 2024-05-08 at 11 05 24" src="https://github.com/Automattic/pocket-casts-android/assets/5845095/e6da28c4-863d-4aec-a8a9-48e5d2971a29">